### ISSUE TITLE
[Question] Dreamer-V3 exploration

### ISSUE THREAD

**Issue by belerico** [Eclectic-Sheep/sheeprl#190]
From @Disastorm:

btw separate question, is there a way to set exploration in dreamerv3? would i adjust ent_coef, or do i need to use one of those other things like the Plan2Explore configs ( I don't know what Plan2Explore is ).

**Comment by belerico** [comment]
I would not suggest to adjust the entropy coefficient because, as the authors wrote in the paper, it should work for multiple environments and actions (in every our experiment we have never changed it). You can instead use [Plan2Explore](https://arxiv.org/abs/2005.05960), which works like this:

1. 1st phase: pre-training with intrinsic reward, i.e. the agent plays to be curious (reach novel states). In this phase two Actor-Critics are learned: the first one plays to be curious, the second one is trained on the extrinsic reward (the one given by the environment) obtained with the explorative actions (this second Actor-Critic is useful if you want to solve your env in a zero-shot manner)
2. 2nd phase: fine-tuning. In this phase everything regarding the world is frozen and the 2nd Actor-Critic trained zero-shot in the 1st phase is finetuned

You can have a look at our [P2E-DV3 README](https://github.com/Eclectic-Sheep/sheeprl/tree/main/sheeprl/algos/p2e_dv3) for more info

**Comment by Disastorm** [comment]
I see, does this plan2explore then use these p2e_dv2_exploration.py and p2e_dv2_finetuning.py instead of the normal sheeprl.py script for training?

Also is there any way I could continue my existing dreamer v3 checkpoint into the plan2explore style, or does plan2explore basically just require restarting my training under this new algorithm?

**Comment by belerico** [comment]
> I see, does this plan2explore then use these p2e_dv2_exploration.py and p2e_dv2_finetuning.py instead of the normal sheeprl.py script for training?

You can run it always with `python sheeprl.py exp=p2e_dv3_exploration ...` and `python sheeprl.py exp=p2e_dv3_finetuning`. Please have a look at both configs at [p2e_dv3_exploration.yaml](https://github.com/Eclectic-Sheep/sheeprl/blob/main/sheeprl/configs/exp/p2e_dv3_exploration.yaml) and [p2e_dv3_finetuning.yaml](https://github.com/Eclectic-Sheep/sheeprl/blob/main/sheeprl/configs/exp/p2e_dv3_finetuning.yaml)

> Also is there any way I could continue my existing dreamer v3 checkpoint into the plan2explore style, or does plan2explore basically just require restarting my training under this new algorithm?

Nope, cause you're missing some models needed for P2E to run. You should start a new experiment

**Comment by Disastorm** [comment]
Ok thanks i think this answers my initial question then.

**Comment by Disastorm** [comment]
@belerico 
I noticed dreamer v3 code has reference to 
cfg.algo.actor.expl_amount
fg.algo.actor.expl_min

are these ways to do exploration without using the p2e_dv3 stuff?

**Comment by belerico** [comment]
Can you please open a new issue while referencing this one?

**Comment by Disastorm** [comment]
Thats ok I just tried it out and I think it worked ( I think it increased exploration ).
